### PR TITLE
Setup rollup

### PR DESCRIPTION
**Pull request by brentongunning**
Sets up rollup with three build flavors:

1. `main` - For use in node applications
2. `browser` - For use in a web browser
3. `module` - For use where es6 is available (ie. linking to this lib from our other libs)

Each is minified. Each goes into the dist folder which will be distributed when we publish now.

**Test Plan**:
- [x] lint, flow test
- [x] test with main and module flavors
- [x] load browser build into chrome and test creating objects

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,13 +3,15 @@
   "version": "0.1.12",
   "description": "A simple, safe, and powerful JavaScript Bitcoin Cash library.",
   "author": "Clemens Ley <ley.clemens@gmail.com>",
-  "main": "src/index.js",
+  "license": "MIT",
+  "main": "dist/bitcoin-source.cjs.min.js",
+  "browser": "dist/bitcoin-source.browser.iife.min.js",
```
Comment (brentongunning) [inline package.json]: remove browser

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import { terser } from 'rollup-plugin-terser'
+
+// The text to appear at the top of the generated files.
+const licenseBanner = `/**
+ * @license
+ * https://github.com/the-bitcoin-token/BitcoinSource
+ * Copyright (c) 2019 Bitcoin Computer
+ * Copyright (c) 2019 Brenton Gunning
+ * Copyright (c) 2018 Janez Urevc
+ * Copyright (c) 2018 Clemens Ley
```
Comment (brentongunning) [inline rollup.config.js]: 2018-2019

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ * Copyright (c) 2011 Google Inc.
+ * Distributed under the MIT software license, see the accompanying
+ * file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+ */`
+
+// Disable circular dependency warnings in our library because they are allowed in ES6 and don't
+// cause problems for us. The maintainer of Rollup does not plan to remove this warning. This is
+// his recommendation. See: https://github.com/rollup/rollup/issues/2271
+function onwarnHook(warning, rollupWarn) {
+  if (warning.code !== 'CIRCULAR_DEPENDENCY') {
```
Comment (brentongunning) [inline rollup.config.js]: Have a list of the specific warnings that don't cause problems

**Comment by ClemensLey** [comment]
LGTM

**Pull request merged**